tdp/cli/commands/deploy.py

```diff
@@ -90,14 +90,15 @@ def deploy(
 
         if dry:
             for _ in deployment_iterator:
-                pass
+                _()
             return
 
         # deployment and operations records are mutated by the iterator so we need to
         # commit them before iterating and at each iteration
-        session.commit()  # Update deployment status to RUNNING, operations to PENDING
```

Comment:
Why did you removed this commit? We should still save initial state to the database.

Reply:
This is done with line beneath the for loop since it commits before each iteration